spec/castle_devise/integration/password_update_spec.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+RSpec.describe "Password update", type: :request do
+  subject(:send_password_update) do
+    put "/users",
+      params: {
+        email: email,
+        user: {
+          current_password: current_password,
+          password: new_password,
+          password_confirmation: new_password
+        },
+        castle_request_token: request_token
+      }
+  end
+
+  let(:facade) { instance_double(CastleDevise::SdkFacade) }
```

Comment:
you have specs that always check whether the facade received or haven't received certain messages. Wouldn't `instance_spy` be better here? This way you can skip some `allow` calls in the before filters.